### ISSUE TITLE
[Opinion]: go2rtc is a mistake

### ISSUE THREAD

**Issue by Majestic7979** [blakeblackshear/frigate#7137]
### Describe the problem you are having

I was using Frigate 0.11 successfully with rtsp-simple-server (now called mediamtx).

It was working fine, not the best, but okay.

Then I installed 0.12 and migrated to its built-in go2rtc. BIGGEST MISTAKE.

I was suffering with constant cutouts of my cameras, and rtc connection that would take several seconds to establish.

So last week, I decided to go back to two separate containers. I removed the go2rtc config from Frigate and put my camera feeds into mediamtx. Then I ingest the mediamtx proxied feeds into frigate.

WOW! Now it's flawlessly streaming. No more cutouts. It takes literally like half a second for a webrtc connection to get established in mediamtx.

I will 100% be critical here (of go2rtc/the code) in the name of user experience and reliability (and not personal attack to be clear) = go2rtc is nowhere near stable and ready for inclusion in something like an NVR and I think it's a mistake that 0.12 introduced it and continues to rely on it. Frigate should focus on Frigate. Now you guys have built in code from someone else into Frigate and that code is clearly not optimized or reliable, since under the exact same system/conditions it runs much worse. Don't believe me just try it with a wide variety of cameras. This is my ffprobe for reference:

```
[{"return_code":0,"stderr":"","stdout":{"programs":[],"streams":[{"avg_frame_rate":"0/0","codec_long_name":"H.264/AVC/MPEG-4AVC/MPEG-4part10","height":1944,"width":2592},{"avg_frame_rate":"0/0","bit_rate":"64000","codec_long_name":"PCMA-law/G.711A-law"}]}}]
```

Please consider unbundling it and going back to basics to focus on the NVR part.

Oh, and mediamtx is handling audio from my cameras just fine, whereas go2rtc couldn't ever do it. Both AAC as well as G711. Frigate's built-in presets aren't processing correctly though, Frigate fails with a black frame for each of the 4 cameras when the audio generic audio preset is used, so I have to record my cctv without audio which isn't ideal. This was happening way back, when frigate was still a new project.

I think this project should focus on stability of the receiving stream instead of packing in projects from others into it and just causing even more issues.

I hope you take this feedback on board and think about it instead of responding defensively. Frigate is great, don't ruin it with things that are buggy. Focusing on the presets and widening compatibility and stability should be primary aims in my opinion. What's the point of a NVR that fails to record, after all? Food for thought.

### Version

0.12

### Frigate config file

```yaml
Ask if required
```

### Relevant log output

```shell
Ask if required
```

### FFprobe output from your camera

```shell
In original post
```

### Operating system

Debian

### Install method

Docker CLI

### Coral version

Other

### Network connection

Wired

### Camera make and model

Anpviz 5MP

### Any other information that may be helpful

Ask if required

**Comment by NickM-27** [comment]
Did you use go2rtc 1.5.0 or just stick with the built in 1.2.0? You can of course use custom versions https://docs.frigate.video/configuration/advanced#custom-go2rtc-version which has many fixes.

There is also no requirement to use go2rtc, you can easily just not use it and it will do nothing and frigate will run the same as before.

Also, go2rtc helps provide a number of essential NVR functions like high quality live streams with low latency. I along with the vast majority of frigate users have been using it without any stability issues at all. 

This was most likely a configuration error.

Finally, support issues are not the correct place for this discussion so this is being closed.

**Comment by NickM-27** [comment]
As an example, based on your ffprobe your cameras have pcm audio. Frigate tells go2rtc to only send audio if it's AAC by default for compatibility reasons with the HA integration. 

The configuring go2rtc guide covers this by suggesting adding an aac audio transcode, but it seems this was not followed in your case. 

The other fix would be to add `?video&audio` to the end of the go2rtc rtsp url to request all video and audio. 

So basically, the audio issues are being blamed on go2rtc for something that is user error and potentially documentation that needs to be improved.

**Comment by Majestic7979** [comment]
Nick let's not talk about making presumptions and then follow up with presumptions? I have asked for help with this in the past - not just with this GH account. For instance, I tried with AAC, and the same issue occurred. It could be anything. You don't know and I don't know. Why blame it on user error if mediamtx is working out of the box? I had go2rtc configured per documentation.

My comment explained what this is about - an opinion. And I said it's about the software not the people making it. You are taking it personal AND making this personal.

The goal of anything that wants to establish itself is to make it as user-friendly, reliable and stable as possible. If Frigate wants to be a niche product fine, let's cut this conversation right here. But if Frigate wants to be anything like the levels of Home Assistant then it will need to do what Home Assistant does which is to make it as easy to use and stable as possible. That means focusing on the application itself. Make a plugin system for other stuff I guess. But including go2rtc (which I also ran separately in a docker container in host mode to avoid any ports issues, by the way) adds an external layer of potential issues that you can't control alone because you don't write the application for that project.

Please listen to feedback, you don't have to accept it but these points are valid. I will not open an issue for config help regarding the audio right now as I've spent far too much time trying to diag and troubleshoot. I'm just happy that video recording is flawless most of the time.

**Comment by blakeblackshear** [comment]
Thanks for your feedback. It's the best way for us to know what users are capable of understanding so we can minimize the time we spend providing support. This project relies on tons of third party dependencies that are maintained by others. Go2rtc is just one of them.

Fortunately for users that don't need their NVR to support RTSP relay or viewing full resolution frame rate live views with audio in the UI, they can simply not configure go2rtc with zero impact. It's totally your choice.

Just like any other dependency, if a better option comes along, we can consider switching to it. So far there have been relatively few support issues related to go2rtc. If that changes, we will reevaluate.

I don't think there is any value in continuing this discussion further.
